src/Playwright.CLI/Program.cs

```diff
@@ -0,0 +1,164 @@
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Threading;
+
+namespace Microsoft.Playwright.CLI
+{
+    class Program
+    {
+        private const string DriverEnvironmentPath = "PW_CLI_DRIVERPATH";
+
+        static void Main(string[] args)
+        {
+            string pwPath = GetFullPath();
+            Console.WriteLine(pwPath);
+            var playwrightStartInfo = new ProcessStartInfo(pwPath, string.Join(' ', args))
+            {
+                UseShellExecute = false,
+                CreateNoWindow = true,
+                RedirectStandardError = true,
+                RedirectStandardInput = true,
+                RedirectStandardOutput = true
+            };
+
+            var pwProcess = new Process()
+            {
+                StartInfo = playwrightStartInfo,
+            };
+
+            playwrightStartInfo.EnvironmentVariables.Add("PW_CLI_TARGET_LANG", "csharp");
+            playwrightStartInfo.EnvironmentVariables.Add("PW_CLI_NAME ", "playwrightcli");
+
+            using var outputWaitHandle = new AutoResetEvent(false);
+            using var errorWaitHandle = new AutoResetEvent(false);
+
+            pwProcess.OutputDataReceived += (_, e) =>
+            {
+                if (e.Data == null) outputWaitHandle.Set();
+                else
+                {
+                    Console.WriteLine(e.Data);
+                }
+            };
+
+            pwProcess.ErrorDataReceived += (_, e) =>
+            {
+                if (e.Data == null) errorWaitHandle.Set();
+                else
+                {
+                    Console.Error.WriteLine(e.Data);
+                }
+            };
+
+            pwProcess.Start();
+
+            pwProcess.BeginOutputReadLine();
+            pwProcess.BeginErrorReadLine();
+
+            pwProcess.WaitForExit();
```

Comment:
[Stackoverflow](https://stackoverflow.com/questions/8808663/get-live-output-from-process) says this is going to block, and so we won't see live updates. Perhaps we should await for `process.Exited` event to unblock the thread so it can perform live streaming of the output? I am also not sure what `WaitOne(5000)` is for?

Reply:
It'll block the calling thread, yes, but it launches the process from a different thread and the callbacks happen on other threads:

![image](https://user-images.githubusercontent.com/1932563/119533211-e9752280-bd85-11eb-8177-3acc7569fa69.png)

The `WaitOne` is to ensure that we don't make the same mistake as our tests did initially - disposing stuff before we get all the content. We either wait for `null` to be sent as the payload, when the stream is done, or 5 seconds after the process exited which likely means there was a problem in the stream.

Reply:
FWIW, the Process implementation [uses](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs#L214) `AsyncStreamReader` underneath, and that queues the work using [this](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/AsyncStreamReader.cs#L247). 

So, it's possible that we get a piece of text dispatched when the call to `WaitForExit`  already returns - hence the `AutoResetEvent` :-).